front/src/modules/simulationResult/components/SpaceTimeChart/utils.ts

```diff
@@ -0,0 +1,6 @@
+/* eslint-disable import/prefer-default-export */
+
+export const getTrainIdsPackage = (tenIndex: number, trainIds: number[], batchSize: number) => {
+  const upperIndex = Math.min(tenIndex * batchSize + batchSize, trainIds.length);
```

Comment:
We don't actually need `Math.min` here: `slice()` takes all remaining elements if `upperIndex > trainIds.length`.

Reply:
Thanks !